components/popover/provider.js

```diff
@@ -1,27 +1,22 @@
 /**
  * External dependencies
  */
-import { noop } from 'lodash';
+import { Slot } from 'react-slot-fill';
 
 /**
  * WordPress dependencies
  */
 import { Component } from '@wordpress/element';
 
 class PopoverProvider extends Component {
```

Comment:
Without context, this doesn't need to be a component class. I also don't think it needs to be a provider in the sense of having children, but rather a slot rendered directly in the layout component. Something like: `<PopoverSlot />`.